core/src/test/scala/org/apache/spark/sql/store/UnifiedPartitionerTest.scala

```diff
@@ -0,0 +1,350 @@
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.ColumnName
+import org.apache.spark.sql.catalyst.expressions.GenericMutableRow
+import org.apache.spark.sql.execution.QueryExecution
+import org.apache.spark.unsafe.types.UTF8String
+
+/**
+ * This test checks the validity of various functionality when we use Spark's partitioner logic for underlying
+ * GemXD storage.
+ */
+class UnifiedPartitionerTest extends SnappyFunSuite
+with Logging
+with BeforeAndAfter
+with BeforeAndAfterAll {
+
+  after {
+    snc.dropTable(RowTableName2, ifExists = true)
+    snc.dropTable(RowTableName1, ifExists = true)
+    snc.dropTable(ColumnTableName2, ifExists = true)
+    snc.dropTable(ColumnTableName1, ifExists = true)
+
+
+  }
+
+  val ColumnTableName1: String = "ColumnTable1"
+  val ColumnTableName2: String = "ColumnTable2"
+  val RowTableName1: String = "RowTable1"
+  val RowTableName2: String = "RowTable2"
+
+  val props = Map.empty[String, String]
+
+
+  val options = "OPTIONS (PARTITION_BY 'col1')"
+
+  val optionsWithURL = "OPTIONS (PARTITION_BY 'Col1', URL 'jdbc:snappydata:;')"
+
+
+  test(" Test hash codes for all Sql types ") {
+    snc.sql(s"CREATE TABLE $ColumnTableName1(OrderId INT ,ItemId INT, ItemRef INT) " +
+        "USING column " +
+        "options " +
+        "(" +
+        "PARTITION_BY 'OrderId'," +
+        "PERSISTENT 'ASYNCHRONOUS')")
+
+    val regtwo: Region[_, _] = Misc.getRegion("/APP/COLUMNTABLE1", true, true)
+    val rattr: RegionAttributes[_, _] = regtwo.getAttributes
+    val pr: PartitionResolver[_, _] = rattr.getPartitionAttributes.getPartitionResolver
+    val rpr: GfxdPartitionByExpressionResolver = pr.asInstanceOf[GfxdPartitionByExpressionResolver]
+    assert(rpr != null)
+
+    // Check All Datatypes
+    val row = new GenericMutableRow(1)
+    var dvd: DataValueDescriptor = new SQLInteger(200)
+    row.update(0, 200)
+
+
+    assert(rpr.getRoutingKeyForColumn(dvd) == row.hashCode)
+
+    row.update(0, new BigInteger("200000"))
+    dvd = new SQLInteger(200000)
+    assert(rpr.getRoutingKeyForColumn(dvd) == row.hashCode)
+
+    row.update(0, true)
+    dvd = new SQLBoolean(true)
+    assert(rpr.getRoutingKeyForColumn(dvd) == row.hashCode)
+
+    row.update(0, new java.sql.Date(1, 1, 2011))
+    dvd = new SQLDate(new java.sql.Date(1, 1, 2011))
+    assert(rpr.getRoutingKeyForColumn(dvd) == row.hashCode)
+
+
+    val ipaddr: Array[Byte] = Array(192.toByte, 168.toByte, 1.toByte, 9.toByte)
+    row.update(0, ipaddr)
+    dvd = new SQLBit(ipaddr)
+    assert(rpr.getRoutingKeyForColumn(dvd) == row.hashCode)
+
+    dvd = new SQLReal(10.5F)
+    row.update(0, 10.5F)
+    assert(rpr.getRoutingKeyForColumn(dvd) == row.hashCode)
+
+    dvd = new SQLLongint(479L)
+    row.update(0, 479L)
+    assert(rpr.getRoutingKeyForColumn(dvd) == row.hashCode)
+
+    dvd = new SQLVarchar("xxxx");
+    row.update(0, UTF8String.fromString("xxxx")) // As catalyst converts String to UtfString
```

Comment:
Let's just use mutable internalrow throughout this test. Then you would have discovered issues with date, timestamp and possibly others I may have missed.